packages/react-server/src/ReactFlightHooks.js

```diff
@@ -91,3 +102,15 @@ export function setCurrentCache(cache: Map<Function, mixed> | null) {
 export function getCurrentCache() {
   return currentCache;
 }
+
+function useId(): string {
+  if (currentRequest === null) {
+    throw new Error('useId can only be used while React is rendering');
+  }
+  const prefix = currentRequest.identifierPrefix
```

Comment:
Instead of checking this every useId call, just set it up to an empty string in createRequest.

Reply:
Would it make sense to go even further and stash the identifierPrefix and identifierCount in module scope alonside the request and then reference prefix / increment count until resetHooks is called and stash the results back on the request?